Dungeoneer/src/com/interrupt/dungeoneer/entities/items/FusedBomb.java

```diff
@@ -9,186 +9,176 @@
-	public float randomCountdownTimer = 50f;
+    @EditorProperty
+    public float randomCountdownTimer = 50f;
 
-	@EditorProperty
-	public float chanceIsDud = 0.125f;
+    @EditorProperty
+    public float chanceIsDud = 0.125f;
 
-	public boolean isDud = false;
+    public boolean isDud = false;
 
-	public boolean wasSpawned = false;
+    public boolean wasSpawned = false;
 
-	@EditorProperty
-	public triggerBehaviorType triggerBehavior = triggerBehaviorType.light;
+    @EditorProperty
+    public triggerBehaviorType triggerBehavior = triggerBehaviorType.light;
 
-	public float explosionRadius = 3f;
-	public float explosionImpulse = 0.2f;
-	public float explosionDamage = 3f;
-	public Weapon.DamageType explosionDamageType = Weapon.DamageType.PHYSICAL;
-	public Color explosionColor = Colors.EXPLOSION;
+    public float explosionRadius = 3f;
+    public float explosionImpulse = 0.2f;
+    public float explosionDamage = 3f;
+    public Weapon.DamageType explosionDamageType = Weapon.DamageType.PHYSICAL;
+    public Color explosionColor = Colors.EXPLOSION;
 
-	private Color flashColor = new Color(Colors.BOMB_FLASH);
+    private final Color flashColor = new Color(Colors.BOMB_FLASH);
 
-	private boolean isWet = false;
+    private boolean isWet = false;
 
-	Explosion explosion = new Explosion();
+    Explosion explosion = new Explosion();
 
-	Array<Entity> spawns = new Array<Entity>();
-	int spawnsCount = 1;
-	public Vector3 spawnVelocity = new Vector3(0.0f, 0.0f, 0.0625f);
-	public Vector3 spawnRandomVelocity = new Vector3(0.125f, 0.125f, 0.0625f);
-	public Vector3 spawnSpread = new Vector3(0.125f, 0.125f, 0.0f);
-	public float spawnMomentumTransfer = 1.0f;
+    Array<Entity> spawns = new Array<>();
+    int spawnsCount = 1;
+    public Vector3 spawnVelocity = new Vector3(0.0f, 0.0f, 0.0625f);
+    public Vector3 spawnRandomVelocity = new Vector3(0.125f, 0.125f, 0.0625f);
+    public Vector3 spawnSpread = new Vector3(0.125f, 0.125f, 0.0f);
+    public float spawnMomentumTransfer = 1.0f;
 
-	public float timerStart = -1f;
+    public float timerStart = -1f;
 
-	public FusedBomb() {
-		super.artType = ArtType.sprite;
-		shadowType = ShadowType.BLOB;
-		collision.set(0.15f, 0.15f, 0.25f);
-		this.name = "Bomb";
-	}
+    public FusedBomb() {
+        super.artType = ArtType.sprite;
+        shadowType = ShadowType.BLOB;
+        collision.set(0.15f, 0.15f, 0.25f);
+        name = "Bomb";
+    }
 
-	@Override
-	public void init(Level level, Level.Source source) {
-		super.init(level, source);
+    @Override
+    public void init(Level level, Level.Source source) {
+        super.init(level, source);
 
-		if(source == Level.Source.SPAWNED && !wasSpawned) {
-			isLit = Game.rand.nextFloat() < this.chanceIsLit;
-			isDud = Game.rand.nextFloat() < this.chanceIsDud;
-		}
-		else if(source == Level.Source.LEVEL_START) {
-			isLit = false;
-			isDud = Game.rand.nextFloat() < this.chanceIsDud;
-		}
+        isLit = Game.rand.nextFloat() < chanceIsLit;
```

Comment:
Won't this now cause the bomb logic to reset on save loads, now that it isn't guarding on `wasSpawned`?

Reply:
Yes, yes you are correct. I've reworked the `init` method:
```java
// Don't set any fields if this was placed in the editor.
if (wasJustPlacedInEditor(source)) {
    return;
}

// Guard against relighting on loading of a save.
if (wasSpawned) {
    return;
}

isLit |= Game.rand.nextFloat() < chanceIsLit;
isDud |= Game.rand.nextFloat() < chanceIsDud;
wasSpawned = true;
```